server/src/main/java/io/seata/server/storage/db/lock/DataBaseDistributedLocker.java

```diff
@@ -0,0 +1,255 @@
+        dbType = configuration.getConfig(ConfigurationKeys.STORE_DB_TYPE);
+        datasourceType = configuration.getConfig(ConfigurationKeys.STORE_DB_DATASOURCE_TYPE);
+
+        if (StringUtils.isBlank(distributeLockTable)) {
+            demotion = true;
+            ConfigurationCache.addConfigListener(DISTRIBUTE_LOCK_DB_TABLE, new ConfigurationChangeListener() {
+                @Override
+                public void onChangeEvent(ConfigurationChangeEvent event) {
+                    String newValue = event.getNewValue();
+                    if (StringUtils.isNotBlank(newValue) && newValue.equalsIgnoreCase(distributeLockTable)) {
+                        distributeLockTable = newValue;
+                        init();
+                        ConfigurationCache.removeConfigListener(DISTRIBUTE_LOCK_DB_TABLE, this);
+                    }
+                }
+            });
+
+            LOGGER.error("The distribute lock table is not config, please create the target table and config it");
+            return;
+        }
+
+        init();
+    }
+
+
+    @Override
+    public boolean acquireLock(DistributedLockDO distributedLockDO) {
+        if (demotion) {
+            return true;
+        }
+
+        Connection connection = null;
+        boolean originalAutoCommit = false;
+        try {
+            connection = distributedLockDataSource.getConnection();
+            originalAutoCommit = connection.getAutoCommit();
+            connection.setAutoCommit(false);
+
+            DistributedLockDO distributedLockDOFromDB = getDistributeLockDO(connection, distributedLockDO.getLockKey());
+            if (null == distributedLockDOFromDB) {
+                boolean ret = insertDistribute(connection, distributedLockDO);
+                connection.commit();
+                return ret;
+            }
+
+            if (distributedLockDOFromDB.getExpireTime() >= System.currentTimeMillis()) {
+                LOGGER.debug("the distribute lock for key :{} is holding by :{}, acquire lock failure.",
+                        distributedLockDO.getLockKey(), distributedLockDOFromDB.getLockValue());
+                connection.commit();
+                return false;
+            }
+
+            boolean ret = updateDistributeLock(connection, distributedLockDO);
+            connection.commit();
+
+            return ret;
+        } catch (SQLException ex) {
+            LOGGER.error("execute acquire lock failure, key is: {}", distributedLockDO.getLockKey(), ex);
+            try {
+                if (connection != null) {
+                    connection.rollback();
+                }
+            } catch (SQLException e) {
+                LOGGER.warn("rollback fail because of {}", e.getMessage(), e);
+            }
+            return false;
+        } finally {
+            try {
+                if (originalAutoCommit) {
+                    connection.setAutoCommit(true);
+                }
+                JdbcUtils.close(connection);
```

Comment:
use IOUtil.close? No need to associate druid.

Reply:
good idea